web/packages/teleterm/src/ui/DocumentAccessRequests/NewRequest/useNewRequest.ts

```diff
@@ -206,7 +207,29 @@ export default function useNewRequest() {
     resourceId: string,
     resourceName?: string
   ) {
-    accessRequestsService.addOrRemoveResource(kind, resourceId, resourceName);
+    if (kind === 'role') {
+      accessRequestsService.addOrRemoveRole(resourceId);
+      return;
+    }
+
+    if (
+      kind === 'app' ||
+      kind === 'db' ||
+      kind === 'kube_cluster' ||
+      kind === 'node'
+    ) {
+      accessRequestsService.addOrRemoveResource(
+        toResourceRequest({
+          kind,
+          resourceId,
+          resourceName,
+          clusterUri,
```

Comment:
I've looked up how `useNewRequest` gets `clusterUri` and I think the way it does this is incorrect. It gets the URI from `localClusterUri` of the workspace, but each `NewRequest` document is bound to a particular cluster AFAIR.

I haven't been able to confirm this as I temporarily cannot log into my local enterprise cluster due to the MFA changes.

Reply:
You are right, we should get the cluster URI from the document.